src/domain/app/App.tsx

```diff
@@ -32,6 +32,31 @@ const instance = createInstance({
   siteId: Number(import.meta.env.REACT_APP_MATOMO_SITE_ID),
 });
 
+const RefreshApiTokenOnLoad: React.FC = () => {
+  const { getUser } = useOidcClient();
+  const { fetch: fetchApiTokens } = useApiTokensClient();
+  const isFetchingStarted = useRef(false);
+
+  useEffect(() => {
+    // Refresh api tokens when page is loaded if the user is already authenticated.
+    // Reason for this is to make sure that expired api token is not used.
+    // That might happen if page is refreshed after refreshing access token. In that case
+    // api token is not refreshed  and might be expired already.
+    // The information about api token expiration time is not stored anywhere.
```

Comment:
Would it be a big effort, considering the release schedule, to implement storing the expiration time to a http only cookie using the Next.js server and checking the value from that here before fetching a new token?

Reply:
Can also be done later, after first releasing this implementation.

Reply:
I personally think that logic to store api token expiration time should be implemented in the HDS Login component where the authentication related logic is implemented. 

Refetching api token in first render is already a hackish way to cover the edge case that access token is refreshed but api token is not. AFAIK Vite server is currently used only to serve static files for the browser and using it for other purposes can take some more time that we have now. 

It would be nice to investigate possibilities to store expiration time at some point but I would skip it for now. I also would like to implement SSR for the UI improve SEO but that's a different story.